### ISSUE TITLE
DTM67-251

### ISSUE THREAD

**Issue by Kotchakorn-2548** [Kotchakorn-2548/Project-Front-end#1]

**Comment by Kotchakorn-2548** [comment]
`import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

from sklearn.model_selection import train_test_split, GridSearchCV
from sklearn.ensemble import RandomForestClassifier
from sklearn.svm import SVC
from sklearn.metrics import classification_report, confusion_matrix
from sklearn.preprocessing import StandardScaler

# 1. จำลอง Dataset ตามโครงสร้างข้อมูลดิบ (95 แถวจาก P01-P22)
np.random.seed(42)
n_samples = 95

# สร้าง Feature จากเซนเซอร์ IMU (Mean, Std, Max, Min ของ IMU Angles/Acc)
X = np.random.randn(n_samples, 6)
# Target Class (1: Correct/ผ่านเกณฑ์, 0: Incorrect/ไม่ผ่านเกณฑ์)
y = np.random.choice([1, 0], size=n_samples, p=[0.93, 0.07])

# 2. แบ่งข้อมูล Train/Test (80:20) แบบ Stratified Split
X_train, X_test, y_train, y_test = train_test_split(
    X, y, test_size=0.20, random_state=42, stratify=y
)

# Scaling
scaler = StandardScaler()
X_train_scaled = scaler.fit_transform(X_train)
X_test_scaled = scaler.transform(X_test)

# 3. Model 1: Random Forest + GridSearchCV
rf_params = {'n_estimators': [10, 50, 100], 'max_depth': [3, 5, 10]}
rf_grid = GridSearchCV(RandomForestClassifier(random_state=42), rf_params, cv=3)
rf_grid.fit(X_train_scaled, y_train)
best_rf = rf_grid.best_estimator_
y_pred_rf = best_rf.predict(X_test_scaled)

# Model 2: SVM + GridSearchCV
svm_params = {'C': [0.1, 1, 10], 'kernel': ['rbf', 'linear']}
svm_grid = GridSearchCV(SVC(random_state=42), svm_params, cv=3)
svm_grid.fit(X_train_scaled, y_train)
best_svm = svm_grid.best_estimator_
y_pred_svm = best_svm.predict(X_test_scaled)

# 4. แสดงผลการทดสอบ
print("==========================================")
print("  ผลการทดสอบโมเดล Random Forest")
print("==========================================")
print(classification_report(y_test, y_pred_rf))

print("==========================================")
print("  ผลการทดสอบโมเดล SVM")
print("==========================================")
print(classification_report(y_test, y_pred_svm))

# พล็อตและเซฟรูป Confusion Matrix
cm = confusion_matrix(y_test, y_pred_rf)
plt.figure(figsize=(6, 4))
sns.heatmap(cm, annot=True, fmt='d', cmap='Blues', 
            xticklabels=['Incorrect (0)', 'Correct (1)'], 
            yticklabels=['Incorrect (0)', 'Correct (1)'])
plt.title('Confusion Matrix - Random Forest')
plt.xlabel('Predicted Class')
plt.ylabel('Actual Class')
plt.tight_layout()
plt.savefig('confusion_matrix.png')
print("\n[SUCCESS] เซฟรูปภาพ 'confusion_matrix.png' เรียบร้อยแล้ว!")
plt.show()`
